src/patterns/molecules/heading/heading.scss

```diff
@@ -0,0 +1,76 @@
+@use "sass:list";
+@use "sass:map";
+@use "../../../core-styling/settings/font";
+@use "../../../core-styling/settings/baseline";
+@use "../../../core-styling/mixins/typography";
+@use "../../../core-styling/mixins/validation";
+
+$_h1-line-height: baseline.$unit-medium;
+$_h2-line-height: baseline.$unit-small + baseline.$unit-smallish;
+$_default_line-height: baseline.$unit-small;
+$_font-family: font.$secondary;
+
+@function _get-scale-point($heading-level) {
+  $scale-point: (
+    "level1": 7,
+    "level2": 4,
+    "level3": 3,
+    "level4": 2,
+    "level5": 1,
+    "level6": 0,
+  );
+  @return map.get($scale-point, $heading-level)
+}
+
+@function _get-line-height($heading-level) {
+  $line-height: (
+    "level1": $_h1-line-height,
+    "level2": $_h2-line-height,
+    "level3": $_default_line-height,
+    "level4": $_default_line-height,
+    "level5": $_default_line-height,
+    "level6": $_default_line-height,
+  );
+  @return map.get($line-height, $heading-level)
+}
+
+@mixin _base-typography {
+  font-family: $_font-family;
+  font-weight: 600;
+}
+
+@mixin _typography($heading-level) {
+  $scale: _get-scale-point($heading-level);
+  $line-height: _get-line-height($heading-level);
+  @include _base-typography();
+  @include typography.set-font-size-and-line-height(baseline.scale($scale), $line-height);
+}
+
+h1 {
```

Comment:
Bit of a different implementation, where the basic styling was done in a global normalization (so you didn't _have_ to use the heading pattern to have a `<h1>` etc).

Think it does make sense to try and do this, but makes we wonder about normalization. (Would we do the same with lists, for example?)